lib/routes/account.js

```diff
@@ -544,19 +541,16 @@ module.exports = function (
         }
 
         function sendVerifyAccountEmail() {
-          // For legacy clients, behavior is to not send an email to verify their account
-          // and have the requestor send it.
-          var sendEmailIfUnverified = false
-
-          // If a value was passed, use that instead
-          if (request.payload.sendEmailIfUnverified !== undefined) {
-            sendEmailIfUnverified = request.payload.sendEmailIfUnverified
-          }
+          // Delegate sending emails for unverified users to auth-server.
+          // Will be removed once all clients have been updated not to send verify emails.
+          // https://github.com/mozilla/fxa-auth-server/issues/1325
+          var shouldSendVerifyAccountEmail = requestHelper.shouldSendVerifyAccountEmail(request)
+          emailSent = false
 
-          var shouldSendVerifyAccountEmail = sendEmailIfUnverified && !emailRecord.emailVerified
           if (shouldSendVerifyAccountEmail) {
             // Resend the account verification email using the tokenVerificationId so that the session
             // that initiated the login will also get verified.
+            emailSent = true
             return mailer.sendVerifyCode(emailRecord, tokenVerificationId, {
```

Comment:
Using the tokenVerificationId is problematic because on line 402 the tokenVerificationId is set to `undefined` if `equestHelper.shouldEnableTokenVerification` returns false. If a `context` is not set, or the context is one that is not on the list that requires token verification, `tokenVerificationId` will not exist, leading to:

> auth-server key server-2 fxa-auth-server.ERROR: request.summary {"op":"request.summary","code":500,"errno":999,"rid":"1473367288601:Shanes-Retina-MBP.local:23385:isusmwid:10002","path":"/v1/account/login","lang":"en,es;q=0.8,de;q=0.6,es-AR;q=0.4,en-US;q=0.2","agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10.11; rv:49.0) Gecko/20100101 Firefox/49.0","remoteAddressChain":["127.0.0.1"],"t":1019,"uid":"00","reason":"signin","email":"a@a.com","stack":"TypeError: Cannot read property 'toString' of undefined\n    at Mailer.mailer.sendVerifyCode (/Users/stomlinson/development/fxa-auth-server/lib/mailer.js:26:23)\n    at sendVerifyAccountEmail (/Users/stomlinson/development/fxa-auth-server/lib/routes/account.js:554:27)\n    at bound (domain.js:287:14)\n    at runBound (domain.js:300:12)\n    at tryCatcher (/Users/stomlinson/development/fxa-auth-server/node_modules/bluebird/js/main/util.js:26:23)\n    at Promise._settlePromiseFromHandler (/Users/stomlinson/development/fxa-auth-server/node_modules/bluebird/js/main/promise.js:507:31)\n    at Promise._settlePromiseAt (/Users/stomlinson/development/fxa-auth-server/node_modules/bluebird/js/main/promise.js:581:18)\n    at Promise._settlePromises (/Users/stomlinson/development/fxa-auth-server/node_modules/bluebird/js/main/promise.js:697:14)\n    at Async._drainQueue (/Users/stomlinson/development/fxa-auth-server/node_modules/bluebird/js/main/async.js:123:16)\n    at Async._drainQueues (/Users/stomlinson/development/fxa-auth-server/node_modules/bluebird/js/main/async.js:133:10)\n    at Immediate.Async.drainQueues [as _onImmediate] (/Users/stomlinson/development/fxa-auth-server/node_modules/bluebird/js/main/async.js:15:14)\n    at processImmediate [as _immediateCallback] (timers.js:383:17)"}